File: src/types/events.ts
import type { ImportSource } from './import'

export interface ScanProgressEvent {
	source: ImportSource
	status: 'scanning' | 'done'
	found: number
}

export type ApplyStep =
	| { kind: 'stoppingSteam' }
	| { kind: 'creatingBackups' }
	| { kind: 'applyingArtwork'; gameName: string | null }
	| { kind: 'updatingShortcuts' }
	| { kind: 'updatingCollections' }
	| { kind: 'restartingSteam' }

export interface ApplyProgressEvent {
	step: ApplyStep
	current: number
	total: number
}

File: src-tauri/src/models/events.rs
use super::importers::ImportSource;
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ScanProgressEvent {
    pub source: ImportSource,
    pub status: String,
    pub found: usize,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase", tag = "kind")]
pub enum ApplyStep {
    StoppingSteam,
    CreatingBackups,
    ApplyingArtwork { game_name: Option<String> },
    UpdatingShortcuts,
    UpdatingCollections,
    RestartingSteam,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ApplyProgressEvent {
    pub step: ApplyStep,
    pub current: usize,
    pub total: usize,
}
